pkg/kv/kvserver/kvflowcontrol/replica_rac2/processor.go

```diff
@@ -58,68 +47,24 @@ type RaftScheduler interface {
-// both Replica mu and raftMu.
-//
-// It should not be essential for read-only methods to hold Replica mu, since
-// except for one case (flushing the proposal buffer), all methods that mutate
-// state in raft.RawNode hold both mutexes. Consider the following information
-// for a replica maintained by the leader: Match, Next, HighestUnstableIndex.
-// (Match, Next) represent in-flight entries, that are not affected by
-// flushing the proposal buffer. [Next, HighestUnstableIndex) are pending, and
-// HighestUnstableIndex *is* affected by flushing the proposal buffer.
-// Additionally, a replica (leader or follower) also has a NextUnstableIndex
-// <= HighestUnstableIndex, which is the index of the next entry that will be
-// sent to local storage (Match is equivalent to StableIndex at a replica), if
-// there are any such entries. That is, NextUnstableIndex represents an
-// exclusive upper bound on MsgStorageAppends that have already been retrieved
-// from Ready. At the leader, the Next value for a replica is <=
-// NextUnstableIndex for the leader. NextUnstableIndex on the leader is not
-// affected by flushing the proposal buffer. RACv2 code limits its advancing
-// knowledge of state on any replica (leader or follower) to
-// NextUnstableIndex, since it is never concerned at any replica with indices
-// that have not been seen in a MsgStorageAppend. This suggests read-only
-// methods should not be affected by concurrent advancing of
-// HighestUnstableIndex.
-//
-// Despite the above, there are implementation details of Raft, specifically
-// maintenance of tracker.Progress, that result in false data races. Due to
-// this, reads done by RACv2 ensure both mutexes are held. We mention this
-// since RACv2 code may not be able to tolerate a true data race, in that it
-// reads Raft state at various points while holding raftMu, and expects those
-// various reads to be mutually consistent.
-type RaftNode interface {
-	rac2.RaftInterface
-	// TermLocked returns the current term of this replica.
-	TermLocked() uint64
-	// LeaderLocked returns the current known leader. This state can advance
-	// past the group membership state, so the leader returned here may not be
-	// known as a current group member.
-	LeaderLocked() roachpb.ReplicaID
-	// LogMarkLocked returns the current log mark of the raft log. It is not
-	// guaranteed to be stablestorage, unless this method is called right after
-	// RawNode is initialized. Processor calls this only on initialization.
-	LogMarkLocked() rac2.LogMark
-	// NextUnstableIndexLocked returns the index of the next entry that will
-	// be sent to local storage. All entries < this index are either stored,
-	// or have been sent to storage.
+// RaftNodeBasicState provides basic state from the RawNode.
+type RaftNodeBasicState struct {
+	// Term is the current term of this replica.
+	Term uint64
+	// Leader is the current known leader. This state can advance past the group
```

Comment:
`Leader is the current known leader, or zero if raft doesn't know the leader. ...`